spring-session-data-mongodb/src/main/java/org/springframework/session/data/mongo/MongoIndexedSessionRepository.java

```diff
@@ -46,15 +47,19 @@
  *
  * @author Jakub Kubrynski
  * @author Greg Turnquist
+ * @author Vedran Pavic
  * @since 2.2.0
  */
 public class MongoIndexedSessionRepository
 		implements FindByIndexNameSessionRepository<MongoSession>, ApplicationEventPublisherAware, InitializingBean {
 
 	/**
 	 * The default time period in seconds in which a session will expire.
+	 * @deprecated since 3.0.0 in favor of
+	 * {@link MapSession#DEFAULT_MAX_INACTIVE_INTERVAL_SECONDS}
```

Comment:
Why does MongoIndexedSessonRepository depend on MapSession?

Reply:
Actually, MongoDB implementation is in many ways an outlier in our codebase. All other session repositories keep their `org.springframework.session.Session` implementations as an internal implementation detail. Also, all those implementations delegate to `MapSession` rather then re-implementing all the logic.